### PR TITLE
Esp32 Compatible Websocket server and Client

### PR DESCRIPTION
**Pull request by williamesp2015**
if hash ported to ESP32 (preliminary exercise:https://github.com/bbx10/Hash_tng/tree/master/src) and resolve functionality of the new ported websockets 
(https://github.com/bbx10/arduinoWebSockets/tree/esp32) it would be great. 
To use PlatformIo it is currently recommended  to use espressif32_stage (http://docs.platformio.org/en/latest/platforms/espressif32.html#using-arduino-framework-with-staging-version) to be able to compile these codes but so far no success in running on ESP32 Dev boards.

### CODE REVIEW

File: src/WebSocketsClient.cpp [hunk 1]
```diff
@@ -117,7 +133,9 @@ void WebSocketsClient::loop(void) {
                 _client.tcp = NULL;
             }
             _client.ssl = new WiFiClientSecure();
+#if !defined(ESP32)
             _client.tcp = _client.ssl;
```
Comment (Links2004) [inline src/WebSocketsClient.cpp]: I think removing this for the ESP32, breaks SSL for it.

File: src/WebSocketsClient.cpp [hunk 2]
```diff
@@ -90,6 +95,17 @@ void WebSocketsClient::beginSSL(const char *host, uint16_t port, const char * ur
...
     beginSSL(host.c_str(), port, url.c_str(), fingerprint.c_str(), protocol.c_str());
 }
+#else
+void WebSocketsClient::beginSSL(const char *host, uint16_t port, const char * url, const char * root_ca_cert, const char * protocol) {
+    begin(host, port, url, protocol);
+    _client.isSSL = true;
+    _root_ca_cert = root_ca_cert;
+}
+
+void WebSocketsClient::beginSSL(String host, uint16_t port, String url, String fingerprint, String protocol) {
```
Comment (Links2004) [inline src/WebSocketsClient.cpp]: i think `fingerprint `is `root_ca_cert` here too?

File: src/WebSocketsClient.h [hunk 3]
```diff
@@ -81,7 +81,11 @@ class WebSocketsClient: private WebSockets {
         uint16_t _port;
 
 #if (WEBSOCKETS_NETWORK_TYPE == NETWORK_ESP8266)
+#if defined(ESP8266)
         String _fingerprint;
+#else
+        String _root_ca_cert;
```
Comment (Links2004) [inline src/WebSocketsClient.h]: where `_root_ca_cert` will used?

File: src/WebSocketsClient.cpp (hunk 1 above)
Comment (williamesp2015) [inline src/WebSocketsClient.cpp]: Intested both websocket server and also websocket client using https://github.com/bbx10/arduinoWebSockets/tree/esp32
I see slower performance compare with ESP8266 with same code and binary frame. Shows intermittent pause inreceiving data from ESP32 to both server and client.

Comment (Links2004) [inline src/WebSocketsClient.cpp]: if 80 is used 8080 is good choice, 81 where working fine too.
Firefox bad port list,
https://dxr.mozilla.org/mozilla-central/source/netwerk/base/nsIOService.cpp#93

may this changes here:
https://github.com/Links2004/arduinoWebSockets/compare/write_big_data
help for the writing.

**Comment by williamesp2015** [comment]
PlatformIo works fine with  espressif32_stage. Please update websocket server and client from 
https://github.com/bbx10/arduinoWebSockets/tree/esp32
I have tested websocket server on ESP32. I suggest to improve websocket sendBIN to be able to send any type of binary array like int16, int32, float,etc.

platform = espressif32_stage
board = esp32dev
framework = arduino

**Comment by Links2004** [comment]
ESP32 support integrated with #265

**Pull request closed without merging**